lib/phoenix/endpoint/cowboy2_websocket.ex

```diff
@@ -0,0 +1,116 @@
+defmodule Phoenix.Endpoint.Cowboy2WebSocket do
+  # Implementation of the WebSocket transport for Cowboy.
+  @moduledoc false
+
+  @behaviour :cowboy_websocket
+  @connection Plug.Adapters.Cowboy2.Conn
+  @already_sent {:plug_conn, :sent}
+
+  def init(req, {module, opts}) do
+    # Is this too brittle?
```

Comment:
Is this a non-rhetorical question? :P

Reply:
Yeah, Is the way I am fetching the config here suitable? In cowboy 2 the timeout is provided here instead of in websocket_init.